runtime/compiler/optimizer/JProfilingValue.cpp

```diff
@@ -331,7 +331,9 @@ TR_JProfilingValue::lowerCalls()
          ipMax = ipLimit;
          }
 
-      if (!stopProfiling && (comp()->getSymRefTab()->getNumInternalPointers() >= ipMax))
+      if (!stopProfiling &&
+          (comp()->getSymRefTab()->getNumInternalPointers() != 0) && // No need to stop profiling if the number of the internal pointers is zero
+          (comp()->getSymRefTab()->getNumInternalPointers() >= ipMax))
```

Comment:
I'm not familiar with where precisely JProfiling ends up using internal pointers.  If it needs to use an internal pointer but both `comp()->maxInternalPointers()` and `comp()->getSymRefTab()->getNumInternalPointers()` were zero, won't this result in the internal pointer's being created and the compilation failing?

Reply:
JProfiling Value it self do not create new internal pointer but they are generated as a part of splitting blocks when it lowers the place holder calls, which would need to uncommon nodes across split points. In case it can not find the register to store the value and reference across split points, for the node, it will create temp slot which could contain internal pointer. So that can increase the `getNumInternalPointers`.
Now if `maxInternalPointer` is set to 0, I expect block splitting to not generate any internal pointer (as there shouldn't be any to begin with). Though if there is such case where JProfiling generates one - I would like to find out more about such case - but it may be safer to stop profiling for now. So original change to check with `>= ipMax` is sufficient.

Reply:
Addressed in [c365d1a](https://github.com/eclipse-openj9/openj9/compare/5cf7d344a9c457270afa1d2f9e2ab60c6d0ee510..c365d1a782aff5cd1412bf99a1d1e598569209c0). Ready for another review. Thanks!